lib/models/consent.dart

```diff
@@ -0,0 +1,236 @@
+class Challenge implements JsonModel {
+  Challenge({this.payload, this.signature});
+
+  @override
+  factory Challenge.fromJson(Map<String, dynamic> json) => _$ChallengeFromJson(json);
+
+  String payload;
+  String signature;
+
+  @override
+  Map<String, dynamic> toJson() => _$ChallengeToJson(this);
+}
+
+@JsonSerializable(explicitToJson: true, includeIfNull: false)
+class TCredentialScope implements JsonModel {
+  TCredentialScope({this.scope, this.accountId});
+
+  @override
+  factory TCredentialScope.fromJson(Map<String, dynamic> json) => _$TCredentialScopeFromJson(json);
+
+  String scope;
+  String accountId;
+
+  @override
+  Map<String, dynamic> toJson() => _$TCredentialScopeToJson(this);
+}
+
+@JsonSerializable(explicitToJson: true, includeIfNull: false)
+class Account implements JsonModel {
+  Account({this.id, this.currency});
+
+  @override
+  factory Account.fromJson(Map<String, dynamic> json) => _$AccountFromJson(json);
+
+  /// Address of the bank account.
+  String id;
+
+  /// Currency of the bank account.
+  Currency currency;
+
+  @override
+  Map<String, dynamic> toJson() => _$AccountToJson(this);
+}
+
+enum TCredentialType {
+  @JsonValue('FIDO')
+  fido,
+}
+
+extension TCredentialTypeJson on TCredentialType {
+  String toJsonString() {
+    return _$TCredentialTypeEnumMap[this];
+  }
+}
+
+enum TCredentialStatus {
+  @JsonValue('PENDING')
+  pending,
+
+  @JsonValue('ACTIVE')
+  active,
+}
+
+extension TCredentialStatusJson on TCredentialStatus {
+  String toJsonString() {
+    return _$TCredentialStatusEnumMap[this];
+  }
+}
+
+enum ConsentStatus {
+  /// Waiting for a callback from Mojaloop to give the consent/account information.
+  @JsonValue('PENDING_PARTY_LOOKUP')
+  pendingPartyLookup,
+
+  /// Waiting for the user to confirm party lookup information and select
+  /// accounts to link
+  @JsonValue('PENDING_PARTY_CONFIRMATION')
+  pendingPartyConfirmation,
+
+  /// Waiting for the user to authorize the account linking.
+  @JsonValue('AUTHENTICATION_REQUIRED')
+  authenticationRequired,
+
+  /// Mojaloop has notified the server that consent has been granted.
+  /// The user has authorized themselves.
+  @JsonValue('CONSENT_GRANTED')
+  consentGranted,
+
+  /// The server has requested that Mojaloop present a challenge
+  /// for the FIDO registration process.
+  @JsonValue('CHALLENGE_GENERATED')
+  challengeGenerated,
+
+  /// The account linking was successful.
+  @JsonValue('ACTIVE')
+  active,
+
+  @JsonValue('REVOKE_REQUESTED')
+  revokeRequested,
+
+  /// The consent was successfully revoked by the user.
+  @JsonValue('REVOKED')
+  revoked,
+}
+
+extension ConsentStatusJson on ConsentStatus {
+  String toJsonString() {
+    return _$ConsentStatusEnumMap[this];
+  }
+}
+
+enum TAuthChannel {
+  @JsonValue('WEB')
+  web,
+
+  @JsonValue('OTP')
+  otp,
+}
+
+extension TAuthChannelJson on TAuthChannel {
+  String toJsonString() {
+    return _$TAuthChannelEnumMap[this];
+  }
+}
+
+enum Currency {
```

Comment:
I think we already have a currency enum under transactions, probably should create a shared one

Reply:
That's true, I can do that